influxdb/csv.go

```diff
@@ -0,0 +1,226 @@
+package influxdb
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	imodels "github.com/influxdata/influxdb/models"
+	"github.com/influxdata/kapacitor/models"
+)
+
+func NewFluxQueryResponse(r io.Reader) (*Response, error) {
+	q := &queryCSVResult{
+		csvReader: csv.NewReader(r),
+		tableNum:  "0",
+		buf:       []imodels.Row{},
+		seriesBuf: &imodels.Row{},
+	}
+	q.csvReader.FieldsPerRecord = -1
+	q.csvReader.ReuseRecord = true
+	return q.response()
+}
+
+// queryCSVResult is the result of a flux query in CSV format
+// it assumes a csv dialect with
+// Annotations: []string{"datatype", "group"},
+// Delimiter:   ",",
+// Header:      true,
+type queryCSVResult struct {
+	csvReader      *csv.Reader
+	Row            []string
+	ColNames       []string
+	colNamesMap    map[string]int
+	dataTypes      []string
+	group          []bool
+	tags           []int
+	measurementCol int
+	fields         []int
+	fieldsLen      int
+	defaultVals    []string
+	Err            error
+	buf            []imodels.Row
+	seriesBuf      *imodels.Row
+	tableNum       string
+}
+
+const skipNonDataFluxCols = 3
+
+func (q *queryCSVResult) response() (*Response, error) {
```

Comment:
Why can't this be re-written as:

```
r := <original io.Reader in NewFluxQueryResponse>
dec := csv.NewMultiResultDecoder(csv.ResultDecoderConfig{})
fluxResultIterator := dec.Decode(r)
for fluxResultIterator.More() {
  res := results.Next()
  if err := res.Tables().Do(func(tbl flux.Table) error {
    // TODO: process flux.Table into Response, see https://pkg.go.dev/github.com/influxdata/flux#Table and https://github.com/influxdata/flux/blob/b9b7f50e098a28318b6dee5032a2da18d51eab9d/execute/format.go#L95
    return err
  }); err != nil {
    return nil, err
  }
}
```

If we can't use the existing parser, let's talk to the flux team about how to get some parts of this parser into the `flux` library - I don't want every application that integrates against flux to need to re-write a parser starting with `csv.Reader`.